src/Foundatio/Messaging/MessageBusBase.cs

```diff
@@ -135,6 +143,13 @@ public abstract class MessageBusBase<TOptions> : IMessageBus, IDisposable where
                 return;
             }
 
+            var subscribers = _subscribers.Values.Where(s => s.IsAssignableFrom(messageType, body)).ToList();
```

Comment:
I see why this was moved below but this really stinks having to do a serialization hit just to check the body. I'd rather we do the subscriber check above and when it gets to this point check the filter if it's not null.

Reply:
You're right. only the number of real subscribers could be different. but i think it would be ok.